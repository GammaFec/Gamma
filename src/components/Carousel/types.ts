export interface ICarousel {
    carouselData: ICarouselData[];
    click: () => void;
    children?: JSX.Element;
}

export interface ICarouselData {
    id: number;
    imageUrl: string;
    imageAlt: string;
    titleOnPage: string;
    text: string;
}
