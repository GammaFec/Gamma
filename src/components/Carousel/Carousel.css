
.carousel{
  display: none;
  transition-duration: 1s ease;

}
.active {
  transition-duration: 1s;
  transform: scale(1.08);
  display: block;
}

