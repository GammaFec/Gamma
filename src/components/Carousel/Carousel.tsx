import React, { useState } from 'react';
import DotSlider from './DotsSlider/DotsSlider';
import Jsondata from './CarouselData.json';

const Carousel = () => {
  const carouselData = Jsondata.CarouselData;

  const [activeTabIndex , setActiveTabIndex] = useState(0)

  const changeActiveTabHandler = (index=0) => {
    console.log(index)
    return setActiveTabIndex(index)
  }



  return (
    <section id='Carousel'>
      <div>{carouselData[activeTabIndex].imageUrl}</div>

      <DotSlider click={changeActiveTabHandler} amountOfDots={carouselData.length} />
    </section>
  );
};

export default Carousel;
