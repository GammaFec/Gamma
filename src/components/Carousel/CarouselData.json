{
  "CarouselData": [
    {
      "imageUrl": "https://Example1.gg/",
      "text": "Sabias que? Casi el 65% de Perros y gatos sufren en las calles de sudamerica. \n Es hora de cambiar eso y darles un hogar!! "
    },
    {
      "imageUrl": "https://Example2.gg/",
      "text": "En esta aplicacion, podras descubir y dar en adopcion a increibles animales listos para hacerte felix y acompanarte en tu hogar."
    },
    {
      "imageUrl": "https://Example3.gg/",
      "text": "Podras conversar con albergues y ademas de ello conocer perritos y gatitos listos para alegrarte el dia.\nRecuerda que ellos son una extension de nosotros y no hay algo mas fiel que ellos. "
    }
  ]
}