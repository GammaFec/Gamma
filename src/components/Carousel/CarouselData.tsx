import CarouselImg01 from "../../img/CarouselImage01.svg";
import CarouselImg02 from "../../img/CarouselImage02.svg";
import CarouselImg03 from "../../img/CarouselImage03.svg";

const data = {
    CarouselData: [
        {
            id: 1,
            imageUrl: CarouselImg01,
            imageAlt: "First Image",
            text:
                "Sabias que? Casi el 65% de Perros y gatos sufren en las calles de sudamerica. \n Es hora de cambiar eso y darles un hogar!! "
        },
        {
            id: 2,
            imageUrl: CarouselImg02,
            imageAlt: "Second Image",
            text:
                "En esta aplicacion, podras descubir y dar en adopcion a increibles animales listos para hacerte felix y acompanarte en tu hogar."
        },
        {
            id: 3,
            imageUrl: CarouselImg03,
            imageAlt: "Third Image",
            text:
                "Podras conversar con albergues y ademas de ello conocer perritos y gatitos listos para alegrarte el dia.\nRecuerda que ellos son una extension de nosotros y no hay algo mas fiel que ellos. "
        }
    ]
};

export default data;
