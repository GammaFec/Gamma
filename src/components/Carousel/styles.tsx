import styled from "styled-components";

import { theme, fontSizes } from "../../common/styles/variables";
import Button from "../Button";

const StyledLiContainer = styled.section<{ active: boolean }>`
    z-index: ${(props): string => (props.active ? "0" : "-10")};
    opacity: ${(props): string => (props.active ? "1" : "0")};
    position: ${(props): string => (props.active ? "relative" : "absolute")};
    top: ${(props): string => (props.active ? "0" : "-30px")};
    transition: 0.5s cubic-bezier(0, 0.84, 0.81, 1.01);

    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
`;

const StyledLi = styled.li`
    overflow: hidden;
    list-style: none;
    width: 100%;
    letter-spacing: -0.44px;

    & > h2 {
        font-size: ${fontSizes.font21};
        margin: 0;
        margin-bottom: 21px;
    }

    & > img {
        width: 70%;
        max-width: 202px;
        height: auto;
        padding: 16px;
        margin-bottom: 11px;
    }

    & > p {
        font-size: ${fontSizes.font12};
        max-width: 272px;
        margin: auto;
    }
`;

const StyledUl = styled.ul`
    display: flex;
    height: 435px;
    font-family: ${theme.fontPrimary};
`;

const StyledButton = styled(Button)`
    width: fit-content;
    margin-right: auto;
    margin-left: auto;
`;

export { StyledLi, StyledUl, StyledLiContainer, StyledButton };
