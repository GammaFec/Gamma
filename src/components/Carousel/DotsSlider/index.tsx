import React from "react";

import { IDotsSlider } from "./types";

import { StyledDot, StyledUl } from "./styles";

const DotsSlider: React.FC<IDotsSlider> = ({
    carouselData,
    click,
    activeTabIndex
}: IDotsSlider) => {
    // const dotsJsx = [];

    // for (let i = 0; i < amountOfDots; i++) {
    //     const newDot = (
    //         <StyledDot key={i} active={activeTabIndex === i} onClick={() => click(i)}></StyledDot>
    //     );
    //     dotsJsx.push(newDot);
    // }
    console.log(carouselData);

    return (
        <StyledUl>
            {carouselData.map((eachObj, index) => {
                return (
                    <StyledDot
                        key={eachObj.id}
                        active={activeTabIndex === index}
                        onClick={() => click(index)}></StyledDot>
                );
            })}
        </StyledUl>
    );
};

export default DotsSlider;
