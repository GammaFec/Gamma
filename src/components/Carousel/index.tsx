import React, { useState } from "react";

//Components

import DotSlider from "./DotsSlider/index";

// Styled Components

import { StyledButton } from "./styles";

//Utils

import { carouselData, carouselDataMapped } from "./utils";

const Carousel: React.FC = () => {
    const [activeTabIndex, setActiveTabIndex] = useState(0);

    const changeActiveTabHandler = (index: number) => {
        return setActiveTabIndex(index);
    };

    return (
        <main id="carousel">
            {carouselDataMapped(activeTabIndex)}
            <DotSlider
                activeTabIndex={activeTabIndex}
                click={changeActiveTabHandler}
                amountOfDots={carouselData.length}
            />
            {activeTabIndex === carouselData.length - 1 ? (
                <StyledButton>Continuar</StyledButton>
            ) : null}
        </main>
    );
};

export default Carousel;
