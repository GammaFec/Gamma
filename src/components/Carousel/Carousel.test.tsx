import React from "react";
import Carousel from "./";
import CarouselData from "./CarouselData";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render } from "@testing-library/react";
import "jest-styled-components";

describe("<Carousel />", () => {
    const mockHandler = jest.fn();

    const carouselData = CarouselData.adopterData;

    test("Render Carousel and childrens", () => {
        const { getByAltText, asFragment } = render(
            <Carousel carouselData={carouselData} handleClickOnContinue={mockHandler} />
        );

        getByAltText("Person with a cat");
        carouselData.forEach(({ imageAlt }) => {
            getByAltText(imageAlt);
        });

        expect(asFragment()).toMatchSnapshot();
    });

    test("Move carousel and click in continue", () => {
        const { getByRole, getAllByRole } = render(
            <Carousel carouselData={carouselData} handleClickOnContinue={mockHandler} />
        );

        const DotButtons = getAllByRole("button");

        carouselData.forEach((_data, i) => {
            DotButtons[i].click();
        });

        const ContinueButton = getByRole("button", { name: "Continuar" });

        expect(ContinueButton).toBeInTheDocument();

        fireEvent.click(ContinueButton);
        expect(mockHandler).toHaveBeenCalledTimes(1);
    });
});
