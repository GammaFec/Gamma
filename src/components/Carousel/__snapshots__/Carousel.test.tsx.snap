// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> Render all items 1`] = `
<DocumentFragment>
  <ul
    class="sc-jSFjdj eXCDtD"
  >
    <section
      class="sc-hKFxyN iByppM"
    >
      <li
        class="sc-eCApnc bUWETU"
      >
        <img
          alt="Person with a cat"
          src="CarouselImage01.svg"
        />
        <h2>
          Elige tu compañero ideal
        </h2>
        <p>
          Aquí el amor a primera vista si existe
        </p>
        <p>
          Descubre tu compañero perfecto y vive muchas historias junto a ellos.
        </p>
      </li>
    </section>
    <section
      class="sc-hKFxyN jrVFHM"
    >
      <li
        class="sc-eCApnc bUWETU"
      >
        <img
          alt="Person with a dog"
          src="CarouselImage02.svg"
        />
        <h2>
          Conversa con Albergues
        </h2>
        <p>
          Heroes sin capa día a día brindan atención, haz match con alguna de las mascotas listas en adopción y define el proceso de adopción.
        </p>
      </li>
    </section>
    <section
      class="sc-hKFxyN jrVFHM"
    >
      <li
        class="sc-eCApnc bUWETU"
      >
        <img
          alt="Person with a bird"
          src="CarouselImage03.svg"
        />
        <h2>
          Recuerda siempre
        </h2>
        <p>
          Cuida, protege y dale amor a quien con solo verte se emociona sin límites.
        </p>
      </li>
    </section>
  </ul>
  <ul
    class="sc-gtsrHT eYqfwW"
  >
    <button
      class="sc-bdnxRM kWgKwT"
    />
    <button
      class="sc-bdnxRM hqtfmP"
    />
    <button
      class="sc-bdnxRM hqtfmP"
    />
  </ul>
</DocumentFragment>
`;
